Include/internal/pycore_stackref.h

```diff
@@ -658,6 +658,16 @@ _Py_TryIncrefCompareStackRef(PyObject **src, PyObject *op, _PyStackRef *out)
 
 #endif
 
+// Like Py_VISIT but for _PyStackRef fields
+#define _Py_VISIT_STACKREF(ref)                                         \
+    do {                                                                \
+        if (!PyStackRef_IsNull(ref)) {                                  \
+            int vret = _PyGC_VisitStackRef(&(ref), visit, arg);         \
```

Comment:
This depends on `_PyGC_VisitStackRef`, which is in `pycore_gc.h`.

What's the motivation for moving the macro here?

Reply:
`pycore_stackref.h` depends on `pycore_object.h` which depends on `pycore_gc.h`.

`_Py_VISIT_STACKREF()` (currently defined in `pycore_gc.h`) uses `PyStackRef_IsNull()` which is defined in `pycore_stackref.h`.

Problem: If dependencies are made explicit (that's part of my work), `pycore_gc.h` should depend on `pycore_stackref.h` which creates a dependency cycle!

Moving `_Py_VISIT_STACKREF()` to `pycore_stackref.h` breaks this cycle.

Reply:
Previously, using `_Py_VISIT_STACKREF()` without including `pycore_stackref.h` worked thanks to luck and the fact that including one header pulled dozens of other headers. With my work on reducing dependencies, this luck is gone, and dependencies must be written explicitly.